contracts/transfer/src/tree.rs

```diff
@@ -57,8 +55,11 @@ impl Tree {
         notes: I,
     ) {
         for note in notes {
-            let leaf = TreeLeaf { block_height, note };
-            self.push(leaf);
+            // skip transparent notes with a value of 0
+            if !note.value(None).is_ok_and(|value| value == 0) {
+                let leaf = TreeLeaf { block_height, note };
```

Comment:
You could leverage [`Iterator::filter`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.filter) here to filter out the notes.